### ISSUE TITLE
Crash on google pay availability check

### ISSUE THREAD

**Issue by manoliskaramanisefood** [Adyen/adyen-android#683]
**Describe the bug**
The following traces are from two crashes on 
`GooglePayComponent.PROVIDER.isAvailable(application, paymentMethod, googlePayConfig)`

According to Google documentation on Google Pay, there should be a try/catch
 
```kotlin
private fun possiblyShowGooglePayButton() {

        val isReadyToPayJson = PaymentsUtil.isReadyToPayRequest() ?: return
        val request = IsReadyToPayRequest.fromJson(isReadyToPayJson.toString()) ?: return

        // The call to isReadyToPay is asynchronous and returns a Task. We need to provide an
        // OnCompleteListener to be triggered when the result of the call is known.
        val task = paymentsClient.isReadyToPay(request)
        task.addOnCompleteListener { completedTask ->
            try {
                completedTask.getResult(ApiException::class.java)?.let(::setGooglePayAvailable)
            } catch (exception: ApiException) {
                // Process error
                Log.w("isReadyToPay failed", exception)
            }
        }
    }
```

but in GooglePayProvider class of the Adyen sdk this try/catch is missing 
```kotlin
val readyToPayTask: Task<Boolean> = paymentsClient.isReadyToPay(readyToPayRequest)
        readyToPayTask.addOnCompleteListener { task ->
            callbackWeakReference.get()?.onAvailabilityResult(task.result == true, paymentMethod, configuration)
        }
```

```
Fatal Exception: com.google.android.gms.tasks.RuntimeExecutionException: com.google.android.gms.common.api.ApiException: 20: The connection to Google Play services was lost due to service disconnection.
       at com.google.android.gms.tasks.zzu.getResult(zzu.java:15)
       at com.adyen.checkout.googlepay.GooglePayProvider.isAvailable$lambda-1(GooglePayProvider.java:82)
       at com.adyen.checkout.googlepay.GooglePayProvider.lambda$OM7waXwkz0OL7VEc3xI6e10MxTQ(GooglePayProvider.java)
       at com.adyen.checkout.googlepay.-$$Lambda$GooglePayProvider$OM7waXwkz0OL7VEc3xI6e10MxTQ.onComplete(-.java:6)
       at com.google.android.gms.tasks.zzj.run(zzj.java:4)
       at android.os.Handler.handleCallback(Handler.java:938)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:236)
       at android.app.ActivityThread.main(ActivityThread.java:7861)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:600)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:967)
```

```
Fatal Exception: com.google.android.gms.tasks.RuntimeExecutionException: com.google.android.gms.common.api.ApiException: 17: API: Wallet.API is not available on this device. Connection failed with: ConnectionResult{statusCode=SERVICE_INVALID, resolution=null, message=null}
       at com.google.android.gms.tasks.zzu.getResult(zzu.java:15)
       at com.adyen.checkout.googlepay.GooglePayProvider.isAvailable$lambda-1(GooglePayProvider.java:82)
       at com.adyen.checkout.googlepay.GooglePayProvider.lambda$OM7waXwkz0OL7VEc3xI6e10MxTQ(GooglePayProvider.java)
       at com.adyen.checkout.googlepay.-$$Lambda$GooglePayProvider$OM7waXwkz0OL7VEc3xI6e10MxTQ.onComplete(-.java:6)
       at com.google.android.gms.tasks.zzj.run(zzj.java:4)
       at android.os.Handler.handleCallback(Handler.java:900)
       at android.os.Handler.dispatchMessage(Handler.java:103)
       at android.os.Looper.loop(Looper.java:219)
       at android.app.ActivityThread.main(ActivityThread.java:8668)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1109)
```

**Comment by jreij** [comment]
Hi, thanks for opening this! We will look into it and update the ticket once we have anything.

**Comment by manoliskaramanisefood** [comment]
Hi! Is this released? Which is the version we should have in our apps to have this fix?

**Comment by jreij** [comment]
Yes it was released yesterday, with [version 4.6.0](https://github.com/Adyen/adyen-android/releases/tag/4.6.0) 🙂
